app/src/main/kotlin/at/bitfire/davdroid/ui/intro/OpenSourceFragment.kt

```diff
@@ -30,18 +58,80 @@ class OpenSourceFragment: Fragment() {
 
 
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
-        val binding = IntroOpenSourceBinding.inflate(inflater, container, false)
-        binding.lifecycleOwner = viewLifecycleOwner
-        binding.model = model
-
-        binding.text.text = getString(R.string.intro_open_source_text, getString(R.string.app_name))
-        binding.moreInfo.setOnClickListener {
-            UiUtils.launchUri(requireActivity(), App.homepageUrl(requireActivity()).buildUpon()
-                    .appendPath("donate")
-                    .build())
+        return ComposeView(requireContext()).apply {
+            setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+            setContent {
+                MdcTheme {
+                    var dontShow by remember { mutableStateOf(model.dontShow.get()) }
+
+                    FragmentContent(
+                        dontShow = dontShow,
+                        onChangeDontShow = {
+                            model.dontShow.set(it)
+                            dontShow = it
+                        }
+                    )
+                }
+            }
         }
+    }
+
 
-        return binding.root
+    @Preview(
+        showBackground = true,
+        showSystemUi = true
+    )
+    @Composable
+    fun FragmentContent(
+        dontShow: Boolean = false,
+        onChangeDontShow: (Boolean) -> Unit = {}
+    ) {
+        val uriHandler = LocalUriHandler.current
```

Comment:
Interesting, didn't know `UriHandler`. It seems to do what we want, but I couldn't find it mentioned on developer.android.com somewhere (only the API). Where did you learn about it? Are there samples where it's used?

I'm also asking because it seems to be implemented by `AndroidUriHandler`, which starts `context.startActivity()`. However it doesn't check whether there's really an available Intent filter and crashes the app when there's no browser available (you can reproduce it by disabling Chrome in the emulator). This case happens from time to time.

Reply:
I think someone suggested it to me, or I just found it out in Android Studio. In any case, we should add a try-catch for `IllegalArgumentException` as indicated in the [reference](https://developer.android.com/reference/kotlin/androidx/compose/ui/platform/UriHandler#openUri(kotlin.String)).

Reply:
Contrary to the documentation, it throws an ActivityNotFoundException, which is not an IllegalArgumentException.

I have added a `SafeAndroidUriHandler` in c8809f0 to have a reusable Compose solution – what do you think of it?

Reply:
Looks really good. I'd use it for now

Reply:
One thing I like to do, for example, when we get rid of accompanist's `MdcTheme`, is using the theme wrapper function to provide the composition providers needed. For example:
```kotlin
fun AppTheme(...) {
  MaterialTheme {
    // provide the uri handler directly
    CompositionLocalProvider(LocalUriHandler provides uriHandler) {
      content()
    }
  }
}
```
Since we are always calling `AppTheme` on every function, we can override this automatically, and just have to call `LocalUriHandler.current` when needed.

But yeah, something to think about in the future.

Reply:
Had the same idea :) Probably when we replace the deprecated MdcTheme …